x-pack/plugins/maps/public/plugin.ts

```diff
@@ -106,14 +113,25 @@ export class MapsPlugin
       MapsPluginSetupDependencies,
       MapsPluginStartDependencies
     > {
+  readonly _initializerContext: PluginInitializerContext<MapsXPackConfig>;
+
+  constructor(initializerContext: PluginInitializerContext<MapsXPackConfig>) {
+    this._initializerContext = initializerContext;
+  }
+
   public setup(core: CoreSetup, plugins: MapsPluginSetupDependencies) {
+    const config = this._initializerContext.config.get<MapsConfigType>();
+    const kibanaVersion = this._initializerContext.env.packageInfo.version;
     const { inspector, home, visualizations, embeddable } = plugins;
-    bindSetupCoreAndPlugins(core, plugins);
+    bindSetupCoreAndPlugins(core, plugins, config, kibanaVersion);
 
     inspector.registerView(MapView);
     home.featureCatalogue.register(featureCatalogueEntry);
     visualizations.registerAlias(getMapsVisTypeAlias());
     embeddable.registerEmbeddableFactory(MAP_SAVED_OBJECT_TYPE, new MapEmbeddableFactory());
+    return {
```

Comment:
So that we could access it in `map_controller.js` on line 100: `const config = _.get(npSetup, 'plugins.maps.config', {});` Having switched to using the NP server, it's now on us to publish map config values for outside use. In this case, "outside use" really just means use in legacy.

Reply:
got it, does this mean the return will go away when all of maps is in NP? Should we add a TODO comment to remember to remove the return?

Reply:
We could, but I could also just remove it in #65079 which I'm working on in parallel where `map_controller.js` goes away